modules/internal/ChapelArray.chpl

```diff
@@ -3211,6 +3211,68 @@ module ChapelArray {
       else
         a.add(e);
   }
+
+  //
+  // BaseSparseDom operator overloads
+  //
+  proc +=(ref sd: domain, inds: [] sd.idxType) where isSparseDom(sd) &&
+    sd.rank==1 {
+
+    if inds.size == 0 then return;
+
+    sd._value.dsiBulkAdd(inds);
+  }
+
+  proc +=(ref sd: domain, inds: [] sd.rank*sd.idxType) where isSparseDom(sd) &&
+    sd.rank>1 {
+
+    if inds.size == 0 then return;
+
+    sd._value.dsiBulkAdd(inds);
+  }
+
+  // Currently this is not optimized for addition of a sparse
+  // This assumes that a domain is sorted by row, which may not always be true.
+  // There could be domains sorted by column instead.
+  // It would be nice if there was a common interface in BaseDom to query
+  // whether a domain was row-major or column-major to distinguish these.
+  // The current implementation is a work-around for this shortcoming.
+  proc +=(ref sd: domain, d: domain)
+  where sd._value: DefaultSparseDom && d.rank==sd.rank && sd.idxType==d.idxType {
```

Comment:
Is it correct that the effect of this where clause is that we don't support a += overload for CS* layouts on the LHS that take a domain on the RHS?   If so, that seems unfortunate...